streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamThread.java

```diff
@@ -93,7 +95,7 @@
      *          |      +-----+-------+      |
      *          +<---- | Partitions  |      |
      *          |      | Assigned (3)| <----+
-     *          |      +-----+-------+      |
+     *          |      +-----+-------+      ^
      *          |            |              |
      *          |            |--------------+
```

Comment:
It seems like this here is the problem...AFAICT this line is just to demonstrate that RUNNING -> RUNNING is valid, but it's tied in to the arrows connecting every other state that actually can't transition to Running. Can we change this to a self-contained loop like you added for PARTITIONS_REVOKED?